src/utils/fs.rs

```diff
@@ -12,8 +12,41 @@ url::define_encode_set! {
     pub FILENAME_ENCODE_SET = [SIMPLE_ENCODE_SET] | { '<', '>', ':', '"', '/', '\\', '|', '?', '*' }
 }
 
+/// If a prefix uses the extended-length syntax (`\\?\`), return the equivalent version without it.
+///
+/// Returns `None` if `prefix.kind().is_verbatim()` is `false`.
+fn strip_verbatim_from_prefix(prefix: &path::PrefixComponent<'_>) -> Option<PathBuf> {
+    let ret = match prefix.kind() {
+        path::Prefix::Verbatim(s) => Path::new(s).to_owned(),
+
+        path::Prefix::VerbatimDisk(drive) => [format!(r"{}:\", drive as char)].iter().collect(),
+
+        path::Prefix::VerbatimUNC(_, _) => unimplemented!(),
+
+        _ => return None,
+    };
+
+    Some(ret)
+}
+
 pub(crate) fn try_canonicalize<P: AsRef<Path>>(path: P) -> PathBuf {
```

Comment:
Can we add some tests for this function?

Reply:
Done. I'm not happy with this part of the code, but a proper `fs::normalize` that doesn't hit the filesystem and doesn't prepend a `\\?\` belongs in an external library.